cloudbrain/core.go

```diff
@@ -140,6 +137,33 @@ func (c *Core) CreateInstance(ctx context.Context, providerName string, attr Cre
 	}, nil
 }
 
+// RemoveInstance creates an instance in the database and queues off the cloud
+// create job in the background.
+func (c *Core) RemoveInstance(ctx context.Context, attr DeleteInstanceAttributes) error {
+	inst, err := c.db.GetInstance(attr.InstanceID)
+	if err != nil {
+		return err
+	}
+
+	_, err = c.db.RemoveInstance(inst)
+	if err != nil {
+		return errors.Wrap(err, "error deleting instance in database")
+	}
+
+	err = c.bb.Enqueue(background.Job{
+		UUID:       uuid.New(),
+		Context:    ctx,
+		Payload:    []byte(attr.InstanceID),
+		Queue:      "remove",
+		MaxRetries: MaxCreateRetries,
+	})
+	if err != nil {
+		return errors.Wrap(err, "error enqueueing 'create' job in the background")
```

Comment:
This error message should be updated to say "…enqueueing 'remove' job…" instead.